### PR TITLE
add activation func to precompile functions

### PR DESCRIPTION
**Pull request by ceyonur**
## Why this should be merged

Adds activation function to precompile functions. Activation Functions denotes that if the underlying precompile function has been activated or not.
If not activated it returns "signature not found" as if this function never exists.

## How this works

Changes new precompile function constructor 

## How this was tested

WIll be tested with manager role

### CODE REVIEW

File: precompile/contract/contract.go [hunk 1]
```diff
@@ -15,13 +15,26 @@ const (
 
 type RunStatefulPrecompileFunc func(accessibleState AccessibleState, caller common.Address, addr common.Address, input []byte, suppliedGas uint64, readOnly bool) (ret []byte, remainingGas uint64, err error)
 
+// ActivationFunc defines a function that is used to determine if a function is active
+// The first return value is whether or not the function is active
```
Comment (darioush) [inline precompile/contract/contract.go:19]:
```suggestion
// The return value is whether or not the function is active
```

**Review by darioush — APPROVED** [review]

File: precompile/contract/contract.go [hunk 2]
```diff
@@ -15,13 +15,26 @@ const (
...
+type ActivationFunc func(AccessibleState) (bool, error)
```
[the remark this answers is not in the document]
Comment (ceyonur) [reply precompile/contract/contract.go]: done

File: precompile/contract/contract.go [hunk 3]
```diff
@@ -77,7 +98,12 @@ func (s *statefulPrecompileWithFunctionSelectors) Run(accessibleState Accessible
 	functionInput := input[SelectorLen:]
 	function, ok := s.functions[string(selector)]
 	if !ok {
-		return nil, suppliedGas, fmt.Errorf("invalid function selector %#x", selector)
+		return nil, suppliedGas, InvalidFunctionErr(selector)
```
Comment (aaronbuchwald) [inline precompile/contract/contract.go:101]: can we get rid of this function and just return the formatted error created by the function?

File: precompile/contract/contract.go [hunk 4]
```diff
@@ -80,5 +101,10 @@ func (s *statefulPrecompileWithFunctionSelectors) Run(accessibleState Accessible
 		return nil, suppliedGas, fmt.Errorf("invalid function selector %#x", selector)
 	}
 
+	// Check if the function is activated
+	if !function.IsActivated(accessibleState) {
+		return nil, suppliedGas, fmt.Errorf("invalid non-activated function selector %#x", selector)
```
Comment (aaronbuchwald) [inline precompile/contract/contract.go:106]: Note: the error message does not change the rules of processing the block itself although the return value when an error is returned does.

**Review by aaronbuchwald — APPROVED** [review]
LGTM

**Pull request merged**